src/ip.py

```diff
@@ -125,11 +126,104 @@ def randomMAC():
     return ":".join(["%02x" % x for x in mac])
 
 
+# Decode ifconfig output
+def parse_ifconfig(res, address):
+    links = []
+    ifindex = 1
+
+    for r in res.split("\n"):
+        if re.match(r"^\w+:", r):
+            if ifindex > 1:
+                links.append(link)
+            (ifname, flags, mtu) = re.findall(r"^(\w+): flags=\d+<(.*)> mtu (\d+)", r)[0]
+            flags = flags.split(",")
+            link = {
+                "ifindex": ifindex,
+                "ifname": ifname,
+                "flags": flags,
+                "mtu": mtu
+            }
+            if "LOOPBACK" in flags:
+                link["link_type"] = "loopback"
+                link["address"] = "00:00:00:00:00:00"
+                link["broadcast"] = "00:00:00:00:00:00"
+            elif "POINTOPOINT" in flags:
+                link["link_type"] = "none"
+            else:
+                link["link_type"] = "unknown"
+            ifindex = ifindex + 1
```

Comment:
I don't think we should be making up ifindex numbers, specially if the underlying network stack already issues those, see `man if_nameindex`.

You can parse those from `ifconfig -a -v`, see for example on my mac the en2 has index 10:
```
...
en2: flags=8963<UP,BROADCAST,SMART,RUNNING,PROMISC,SIMPLEX,MULTICAST> mtu 1500 index 10
...
```

Secondly, if we have the correct indexes we should then print them in the relevant human readable output, similar to Linux behavior for `ip link show`, e.g.:

```
...
6: br0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdis
...
```

Reply:
Nice, I completely miss the additional information from -v option of `ifconfig`.

Reply:
Done